### DISCUSSION TITLE
Renaming the main branch

### DISCUSSION THREAD

**Discussion by DavidBrunow** [fastlane/fastlane#18025] [Ideas]
A simple search did not return any results so I'm asking here -- has there been any discussion around renaming the main branch to `main`?

If not I'd like to propose that change now that GitHub supports renaming branches although I do not know the full impact of such a change.

**Comment by Cyberbeni** [comment]
Naming of things is not the real problem, pretending to solve the problem is worse than not doing anything. I've only seen white people push this renaming bullshit while I've only seen opinions like this from Black people: https://twitter.com/johnlegend/status/1276892001208655875

If you want to help solve the problem, send a letter to your political representative, requesting them to make racist symbols (nazi stuff, confederate flag, etc.) illegal.

**Comment by Cyberbeni** [comment]
The term for the remote's HEAD is "default branch", so if you are referring to it, either use the name that is currently used (master) or use the term for it (default). Asking to rename the main branch is nonsense when there is no branch named "main"

**Comment by DavidBrunow** [comment]
Naming of things matters too and my efforts for equality outside of this are completely separate matters. Why are you opposed to the change? Do you think that leaving it as-is moves us towards equality?

**Reply by Cyberbeni** [reply]
https://en.wikipedia.org/wiki/Red_herring

People who are not the cause of issues still like to give a little help towards solving the issue but once they did that, they will just forget about the problem, thinking they already did their part. By promoting non-issues, like branch naming, you are convincing people to help "solve" the issue by doing this and after saying no to other 20 non-issues, they will also say no to an actual issue when someone asks them to help.

Also the name change caused even more issues because people learning how to use git get confused about not finding the master branch. ( https://www.reddit.com/r/git/comments/krp2iy/couldnt_find_remote_ref_master/ )

**Reply by DavidBrunow** [reply]
Here's my take -- if the way that something is named has the potential to exclude a group of the community then I'm going to do what I can to not use that name. And as part of this community, as both a user of Fastlane and (future) contributor (once my outstanding pull requests get merged), I think it is important that topics like this are brought up and discussed in the open.

I don't think this is a red herring because from what I've seen folks do notice the current uses of `master` in technology and that has a negative effect on their involvement or their desire to be involved with technological careers or communities. And in my opinion, if even one person is excluded because of such a trivial thing as a name of a branch then that change needs to be made.

Confusing folks is a valid concern but one that in my opinion is worth the tradeoff to not exclude folks.

**Reply by abardallis** [reply]
And for those that de feel excluded by the name master, we'll be sure to point them to that John Legend tweet so they can be properly educated on how they should feel. I understand what you're saying about "solving the real problem" but you seem to not be recognizing that there are people in our development communities that are bothered by this and welcome the change. 

As for developers getting confused, I would offer that this will actually be worsened by not renaming the branch. Given that Github agrees with the name change, (https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/), repos with a master branch are on the road to becoming less common. Documentation, tutorials, etc. will soon follow suit all causing a good deal of confusion for anyone who starts new work on a repo with a default branch named master.

In the meantime, removing terms from our workflows that have the potential to cause pain to valued members of our development community far outweighs the couple extra minutes it takes someone new to git to google why they're getting a "fatal: couldn't find remote ref master" error.
